### PR TITLE
Doc 806 split environment variables page

### PR DESCRIPTION
**Pull request by ayatnkw**
## Motivation 
- separate pages will show up better in search and potentially be more browsable
- it allows removing the TOC which improves table space 

## Description 
- created a directory named `environment-variables` and split the [environment variable reference page](https://docs.n8n.io/hosting/configuration/environment-variables/) into a page per table
- added an index page
- added brief descriptions for each table 
- removed TOC and added css to display tables at full width of the page
- fixed links that linked to `hosting/configuration/environment-variables/#...` 

## Tickets
[Linear](https://linear.app/n8n/issue/DOC-806/split-environment-variables-page)

### CODE REVIEW

File: docs/_extra/css/extra.css [hunk 1]
```diff
@@ -174,6 +174,14 @@
   .md-typeset table:not([class]) {
   font-size: 0.75rem
   }
+
```
Comment (StarfallProjects) [inline docs/_extra/css/extra.css:177]: This shouldn't be needed. When you hide the TOC, the table should automatically expand, as it has for example here: https://docs.n8n.io/code/builtin/output-other-nodes/

Did you find it wasn't working without the CSS?

**Comment by StarfallProjects** [comment]
Regarding file names: don't include 'filler' words like "and" or "the". For more information on creating good, SEO-friendly slugs: https://www.seo.com/basics/glossary/url-slug/

File: docs/hosting/configuration/environment-variables/database.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+tags:
+  - environment variables
+hide:
+  - toc
+  - tags
+---
+
+# Database environment variables
+
+By default, n8n uses **SQLite** to save credentials, past executions, and workflows. n8n also supports **PostgresDB** but has [removed support for MySQL and MariaDB](/docs/1-0-migration-checklist). This section outlines environment variables to configure your chosen database for self-hosted n8n instance.
```
[the remark this answers is not in the document]
Comment (StarfallProjects) [reply docs/hosting/configuration/environment-variables/database.md:14]: To understand this feedback, please do the following:

1. Go to [Hemingway App](https://hemingwayapp.com/) and copy in your original text:

```
By default, n8n uses SQLite to save credentials, past executions, and workflows. n8n also supports PostgresDB but has removed support for MySQL and MariaDB. This section outlines environment variables to configure your chosen database for self-hosted n8n instance.
```

2. In another tab, open a second instance of Hemingway and copy in:

```
By default, n8n uses SQLite. n8n also supports PostgreSQL. n8n removed support for MySQL and MariaDB in v1.0. 

This section lists environment variables to configure your self-hosted n8n database.
```

You should be able to see the difference in reading grade.

There are a few of other points here:

1. PostgreSQL is the official brand name. Make sure to match it.
2. It's helpful to the user to be specific about when we removed support for the other databases.
3. n8n uses SQLite for everything, not just workflows, creds, and executions. You can see an outline of the database contents here: https://docs.n8n.io/hosting/architecture/database-structure/
4. Make sure to include the final `/` on URLs. If you check the preview, you will see the link currently 404s: https://deploy-preview-1991--n8n-docs.netlify.app/hosting/configuration/environment-variables/database/
5. The database names don't need to be bold. Bold is almost exclusively for UI text.

**Comment by StarfallProjects** [comment]
You will need to check all the URLs: I think many of them are missing the `/` and will 404.

**Comment by StarfallProjects** [comment]
Please go through and rephrase any places where you've used "allows". We have a ticket pending to do this for the whole site, as it's a best practice. For this PR, let's just make sure we're not adding new ones. 

You can see more context on the ticket: https://linear.app/n8n/issue/DOC-631/squish-the-allows

**Comment by StarfallProjects** [comment]
On individual pages, say "This page outlines . . . " rather than "This section outlines . . . "

Consider changing "outlines" to "lists". As an exercise: what do you see as the pros and cons of this change?

File: docs/hosting/configuration/environment-variables/endpoints.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+tags:
+  - environment variables
+hide:
+  - toc
+  - tags
+---
+
+# Endpoints environment variables
+
+This section outlines environment variable configurations for setting payload size limits, enabling and customizing metrics, managing REST and webhook endpoints, and providing options for proxy compatibility.
```
Comment (StarfallProjects) [inline docs/hosting/configuration/environment-variables/endpoints.md:14]: Try this one in https://hemingwayapp.com/ As you'll see it's a very high reading grade. Hemingway suggests aiming for 9 or below. In docs I usually try to go even lower. Play around in Hemingway and see if you can rephrase this.

File: docs/hosting/configuration/environment-variables/external-hooks.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+tags:
+  - environment variables
+hide:
+  - toc
+  - tags
+---
+
+# External hooks environment variables
+
+You can define external hooks that n8n executes whenever a specific operation runs. Refer to [Backend hooks](/embed/configuration/#backend-hooks) for examples of available hooks and [Hook files](/embed/configuration/#hook-files_1) on file formatting.
```
Comment (StarfallProjects) [inline docs/hosting/configuration/environment-variables/external-hooks.md:14]:
```suggestion
You can define external hooks that n8n executes whenever a specific operation runs. Refer to [Backend hooks](/embed/configuration/#backend-hooks) for examples of available hooks and [Hook files](/embed/configuration/#hook-files_1) for information on file formatting. 
```

File: docs/hosting/configuration/environment-variables/executions.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+tags:
+  - environment variables
+hide:
+  - toc
+  - tags
+---
+
+# Executions environment variables
+
+This section outlines environment variables to configure settings related to workflow executions.
```
Comment (StarfallProjects) [inline docs/hosting/configuration/environment-variables/executions.md:14]:
```suggestion
This page lists environment variables to configure workflow execution settings.
```

File: docs/hosting/configuration/environment-variables/logs.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+contentType: reference
+tags:
+  - environment variables
+hide:
+  - toc
+  - tags
+---
+
+# Logs environment variables
+
```
Comment (StarfallProjects) [inline docs/hosting/configuration/environment-variables/logs.md:13]: Add intro sentence

File: docs/hosting/configuration/environment-variables/queue-mode.md [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+tags:
+  - environment variables
+hide:
+  - toc
+  - tags
+---
+
+# Queue mode environment variables
+
+You can run n8n in different modes depending on your needs, and the queue mode provides the best scalability. Refer to [Queue mode](/hosting/scaling/queue-mode/) for its configuration details.
```
Comment (StarfallProjects) [inline docs/hosting/configuration/environment-variables/queue-mode.md:14]:
```suggestion
You can run n8n in different modes depending on your needs. Queue mode provides the best scalability. Refer to [Queue mode](/hosting/scaling/queue-mode/) for more information.
```

File: docs/_extra/css/extra.css (hunk 1 above)
Comment (ayatnkw) [reply docs/_extra/css/extra.css:177]: I noticed that the table width in the `enviornment-variables/database` page would be inconsistent without this css, but maybe I'm missing something here.

Comment (StarfallProjects) [reply docs/_extra/css/extra.css:177]: Agreed: leave it in. Add a comment to say what you've done.

Comment (ayatnkw) [reply docs/_extra/css/extra.css:177]: Reference: [Force data table to use full width #3530](https://github.com/squidfunk/mkdocs-material/discussions/3530) from discussions in mkdocs-material 
Added a comment in the css file.

File: docs/hosting/configuration/environment-variables/endpoints.md (hunk 3 above)
Comment (ayatnkw) [reply docs/hosting/configuration/environment-variables/endpoints.md]: Shortened & simplified :)

File: docs/data/binary-data.md [hunk 8]
```diff
@@ -48,8 +48,8 @@ You can use the [Code node](/code/code-node/) to manipulate binary data in your
...
+You can configure how your self-hosted n8n instance handles binary data using the [Binary data environment variables](/hosting/configuration/environment-variables/binary-data). This includes tasks such as setting the storage path and choosing how to store binary data.
 
 Your configuration affects how well n8n scales: [Scaling | Binary data filesystem mode](/hosting/scaling/binary-data/).
 
-Reading and writing binary files can have security implications. If you want to disable reading and writing binary data, use the `NODES_EXCLUDE` environment variable. Refer to [Environment variables | Nodes](https://docs.n8n.io/hosting/configuration/environment-variables/#nodes) for more information.
+Reading and writing binary files can have security implications. If you want to disable reading and writing binary data, use the `NODES_EXCLUDE` environment variable. Refer to [Environment variables | Nodes](https://docs.n8n.io/hosting/configuration/environment-variables/nodes) for more information.
```
Comment (StarfallProjects) [inline docs/data/binary-data.md:55]:
```suggestion
Reading and writing binary files can have security implications. If you want to disable reading and writing binary data, use the `NODES_EXCLUDE` environment variable. Refer to [Environment variables | Nodes](/hosting/configuration/environment-variables/nodes/) for more information.
```

**Review by StarfallProjects — CHANGES_REQUESTED** [review]

File: docs/hosting/configuration/configuration-examples/index.md [hunk 9]
```diff
@@ -6,8 +6,8 @@ contentType: overview
 
 # Configuration examples
 
-This section contains examples for how to configure n8n to solve particular use cases.
+This page lists examples for how to configure n8n to solve particular use cases.
```
[the remark this answers is not in the document]
Comment (StarfallProjects) [reply docs/hosting/configuration/configuration-examples/index.md:9]: I realise I was unclear in my previous feedback: I didn't want you to remove all reference to sections. Just to make sure you only talked about the section on overview pages, and then the individual page when on a specific page.

**Review by StarfallProjects — APPROVED** [review]

**Pull request merged**